openssl/src/pkcs5.rs

```diff
@@ -138,6 +138,155 @@ pub fn scrypt(
     }
 }
 
+/// Extract a pseudorandom key from an input keying material using the
+/// HKDF-Extract algorithm.
+///
+/// Requires OpenSSL 1.1.1 or newer.
+#[cfg(any(ossl111))]
```

Comment:
Are these two sequences of calls the only set of things you'd ever need to do when interacting with the HKDF API? If not, it probably makes more sense to add type(s) like `Signer` to provide more flexibility.

Reply:
Yes, those two are the only possible call sequences. Note that I put them under `openssl::pkcs5` just because the HKDF operations are similar to the existing `pbkdf2` interface and they have nothing to do with PKCS#5. Perhaps it might make more sense to move them to a dedicated module where SSH KDF etc. can also be supported.

Reply:
Isn't there minimally a third call sequence using EXTRACT_AND_EXPAND? https://www.openssl.org/docs/man1.1.1/man3/EVP_PKEY_CTX_add1_hkdf_info.html

Reply:
By the definition of HKDF ([RFC 5869](https://tools.ietf.org/html/rfc5869)), there are only two functions (HKDF-Extract and HKDF-Expand). OpenSSL provides a combined operation for convenience but it can be easily performed with the HKDF-Extract and HKDF-Expand operations.

Reply:
It seems strange to not expose the thing that OpenSSL's documentation states is the default mode.

Reply:
Makes sense; I've added `hkdf` for that behavior.